src/leap/bitmask/services/mail/imap.py

```diff
@@ -70,3 +72,24 @@ def start_imap_service(*args, **kwargs):
         log.startLogging(sys.stdout)
 
     return imap.run_service(*args, **kwargs)
+
+
+def start_incoming_mail_service(keymanager, soledad, imap_factory, userid):
+    """
+    Initalizes and starts the incomming mail service.
+
+    :returns: a Deferred that will be fired with the IncomingMail instance
+    """
+    def setUpIncomingMail(inbox):
+        incoming_mail = IncomingMail(
+            keymanager,
+            soledad,
+            inbox,
+            userid)
+        incoming_mail.start_loop()
+        return incoming_mail
+
+    # XXX: do I really need to know here how to get a mailbox??
+    d = imap_factory.getMailbox(INBOX_NAME)
```

Comment:
I think the best next step now is to make Mail a composite service, made out of ServiceIncoming, IMAPService and SMTPService, and hiding all initialization complexity from the eyes of the client backend.